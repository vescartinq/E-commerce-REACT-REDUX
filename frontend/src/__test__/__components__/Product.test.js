import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import Product from './../../components/Product';

describe('Product', () => {
  let wrapper;

  beforeEach(() => {
    const product = {
      _id: 1,
      image: 'https:://localhost/algo.jpg',
      name: 'algo',
      rating: 1,
      numReviews: 1,
      price: 1,
    };

    wrapper = shallow(<Product product={product} />);
  });

  test('should render <Product/> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
