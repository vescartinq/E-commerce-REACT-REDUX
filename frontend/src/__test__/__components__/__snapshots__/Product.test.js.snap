// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should render <Product/> component 1`] = `
<div
  className="card"
  key="1"
>
  <Link
    to="/product/1"
  >
    <img
      alt="algo"
      className="medium"
      src="https:://localhost/algo.jpg"
    />
  </Link>
  <div
    className="card-body"
  >
    <Link
      to="/product/1"
    >
      <h2>
        algo
      </h2>
    </Link>
    <Rating
      numReviews={1}
      rating={1}
    />
    <div
      className="price"
    >
      1
      ,00 â‚¬
    </div>
  </div>
</div>
`;
